frontend/src/services/experiment.editor.ts

```diff
@@ -112,30 +116,35 @@ export class ExperimentEditor extends Service {
   }
 
   @computed get currentStage() {
-    return this.stages.find(stage => stage.id === this.currentStageId);
+    return this.stages.find((stage) => stage.id === this.currentStageId);
   }
 
   hasStageKind(kind: StageKind) {
     return this.stages.findIndex((stage) => stage.kind === kind) !== -1;
   }
 
   updateStage(newStage: StageConfig) {
-    const index = this.stages.findIndex(stage => stage.id === newStage.id);
+    const index = this.stages.findIndex((stage) => stage.id === newStage.id);
     if (index >= 0) {
       this.stages[index] = newStage;
     }
   }
 
   setStages(stages: StageConfig[]) {
-    this.stages = stages;
+    for (let stage of stages) {
```

Comment:
set this.stages = [] first! I can push a fix for this

Reply:
Also, my bad - I realized we actually need to do the mustWait whenever stages are edited as well. Pushing a fix for this.